src/esp/sim/Simulator.cpp

```diff
@@ -72,6 +72,13 @@ void Simulator::close(const bool destroy) {
 
   resourceManager_ = nullptr;
 
+  if (debugLineRender_) {
+    // Python may keep around other shared_ptrs to this object, but we need
+    // to release GL resources here.
+    debugLineRender_->releaseGLResources();
+    debugLineRender_ = nullptr;
```

Comment:
This is weird. What I really want to do is delete the DebugLineRender instance here. But I can't because I had to share `shared_ptr`s with Python when doing the bindings. And Python tends to keep around the pointers indefinitely. So we don't actually control when this object gets deleted.

Reply:
We used a weak_ptr wrapper paradigm to get around this with ManagedObjects. The wrapper does a lock and forwards the command if it succeeds, otherwise warning about the stale reference.

Reply:
The `weak_ptr` solution is interesting but feels like overkill for my situation. If a Python user keeps a reference to DebugLineRender after the sim has been closed, and she tries to use it, nothing bad happens.